Domain/Logic/Deserialize/Features.cs

```diff
@@ -0,0 +1,216 @@
+                    batch.Do(x => x.For(service));
+                    features.AddRange(batch);
+                }
+                catch (Exception ex)
+                {
+                    Log.For(typeof(Features)).Warning(ex.ToString());
+
+                }
+            }
+            await Repository.Write("/features/features.json", JsonConvert.SerializeObject(features));
+            Feature.All = await LoadFeatures();
+        }
+        internal static async Task<Feature[]> LoadFeatures()
+        {
+            var json = "";
+            try
+            {
+                json = await Repository.Read("/features/features.json");
+            }
+            catch (Exception ex)
+            {
+                Log.For(typeof(Feature)).Error(ex.ToString());
+                await RefreshServiceFeatures();
+            }
+            if (json.HasValue())
+            {
+                var featureDefinitions = JsonConvert.DeserializeObject<FeatureDefinition[]>(json);
+                return GetActualFeatures(featureDefinitions);
+            }
+            return Feature.All.ToArray();
+        }
+        static Feature[] GetActualFeatures(FeatureDefinition[] featureDefinitions)
+        {
+            var features = new List<Feature>();
+            foreach (var defenition in featureDefinitions)
+            {
+                Feature parent = null;
+                foreach (var title in defenition.FullPath.Split("/"))
+                {
+                    var feature = features.FirstOrDefault(x => x.Title.CompareTo(title) == 0);
+                    if (feature != null)
+                    {
+                        parent = feature;
+                        continue;
+                    }
+                    var fullPath = title;
+                    if (parent != null) fullPath = parent.GetFullPathSlashSeperated() + "/" + title;
+                    var subfeatureDefenition = featureDefinitions.FirstOrDefault(x => x.FullPath.CompareTo(fullPath) == 0);
+                    if (defenition.FullPath.ToLower().EndsWith(title.ToLower())) feature = CreateFeature(defenition, parent);
+                    else if (subfeatureDefenition != null) feature = CreateFeature(subfeatureDefenition, parent);
+                    else
+                        feature = CreateFeature(
+                            new FeatureDefinition
+                            {
+                                FullPath = fullPath,
+                                Icon = "fas fa-folder",
+                                ServiceName = "Hub"
+                            }, parent);
+                    features.Add(feature);
+                    if (parent != null) parent.Children.Append(feature);
+                    parent = feature;
+                }
+            }
+            return features.ToArray();
+        }
+        static Feature CreateFeature(FeatureDefinition featureDefenition, Feature parent)
```

Comment:
Can become a method inside FeatureDefinition named ToFeature(Feature parent)
and move it to a file